apps/CMakeLists.txt

```diff
@@ -48,10 +48,7 @@ target_include_directories(volk_profile
 )
 
 if(NOT FILESYSTEM_FOUND)
-    target_include_directories(volk_profile
-        PUBLIC ${Boost_INCLUDE_DIRS}
-    )
-    target_link_libraries(volk_profile PRIVATE ${Boost_LIBRARIES})
+    target_link_libraries(volk_profile PRIVATE Boost::filesystem)
```

Comment:
Still need to use the `Boost_INCLUDE_DIRS` for includes here

Reply:
`target_link_libraries` with a config target does add the includes to the target, or am I mistaken?

For example, gnuradio/gr-uhd/lib/CMakeLists.txt: https://github.com/gnuradio/gnuradio/blob/master/gr-uhd/lib/CMakeLists.txt#L34

We `target_link_library` the `gnuradio-uhd` target against the UHD::UDH target, which publicly says "hey, include this", which leads to the target automagically gaining these include dependencies.